docs/source/configuration.rst

```diff
@@ -356,18 +356,42 @@ layout: :ref:`layoutconfig`.
 
 .. _templateconfig:
 
-Jinja Templating Configuration
-------------------------------
+Templating Configuration
+------------------------
 
-When thinking about Jinja templating there are two different kinds of things
-that a user might want to fill into a templated file, *variables* and
-*functions/macros*. Currently *functions* aren't implemented in any
-of the templaters.
+This section explains how to configure templating for SQL files.
 
-Variable Templating
-^^^^^^^^^^^^^^^^^^^
+When writing SQL files, users might utilise some kind of templating.
+The SQL file itself is written with placeholders which get rendered to proper
+SQL at run time.
+This can range from very simple placeholder templating to complex Jinja
+templating.
 
-Variables are available in the *jinja*, *python* and *placeholder* templaters.
+SQLFluff supports templated sections in SQL, see :ref:`templater`.
+This is achieved by the following set of operations:
+
+1. SQLFluff pre-renders the templated SQL
+2. SQLFluff applies the lint and fix operations to the rendered file
```

Comment:
This would be a great moment to add a note saying something like: 
> SQLFluff linting requires an understanding of the SQL syntax, which means linting relies on having a valid SQL file to parse. The implication of this for the templater is that sections of the raw file which aren't included in the rendered file (e.g. because they're in an :code:`{% if ... %}` statement), will _not be linted_. This is functionality we hope to support in future.

Whether that's in as a note, or as part of point 2 here I think doesn't particularly matter.

Reply:
Yeah, good idea. I added a note similar to what you suggested.